manifests/config.pp

```diff
@@ -35,13 +35,15 @@
     mode  => '0640',
   }
 
-  $uuid_fact_signature = foreman::hmac_signature($foreman::oauth_consumer_secret, $foreman::instance_id)
-  file { 'foreman_uuid.json':
-    path    => '/opt/puppetlabs/facter/facts.d/foreman_uuid.json',
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0440',
-    content => to_json({'foreman_uuid' => $foreman::instance_id, 'foreman_uuid_signature' => $uuid_fact_signature}),
+  if $foreman::expose_instance_id {
+    $uuid_fact_signature = foreman::hmac_signature($foreman::oauth_consumer_secret, $foreman::instance_id)
+    file { 'foreman_uuid.json':
```

Comment:
Instead of an `if`, I prefer to explicitly manage it:
```suggestion
    file { 'foreman_uuid.json':
      ensure => bool2str($foreman::expose_instance_id, 'file', 'absent'),
```

That allows users to clean it up again.